schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/users/DefaultUserService.java

```diff
@@ -156,102 +161,133 @@ private void notify(User user) {
 
     @Override
-    public User setPassword(User user, String newPassword) throws EntityNotFoundException {
-        Transaction tx = Datastore.beginTransaction();
-        User db = Datastore.getOrNull(tx, userMeta, user.getId());
-        if (db == null) {
-            tx.rollback();
-            throw new EntityNotFoundException();
+    public User setPassword(final User user, final String newPassword) throws EntityNotFoundException {
+        try {
+            return TransactionOperator.execute(new ModelOperation<User>() {
+                @Override
+                public User execute(com.google.appengine.api.datastore.Transaction tx) throws EntityNotFoundException {
+                    User db = Datastore.getOrNull(tx, userMeta, user.getId());
+                    if (db == null) {
+                        throw new EntityNotFoundException();
+                    }
+                    db.setPassword(new ShortBlob(DigestUtil.encrypt(newPassword)));
+                    Datastore.put(tx, db);
+                    tx.commit();
+                    return db;
+                }
+            });
+        } catch (EntityNotFoundException e) {
+            throw e;
+        } catch (ModelException e) {
+            throw Throwables.propagate(e);
         }
-        db.setPassword(new ShortBlob(DigestUtil.encrypt(newPassword)));
-        Datastore.put(tx, db);
-        tx.commit();
-        return db;
     }
 
     @Override
-    public UserLogin addLogin(User user, UserLogin login) throws EntityNotFoundException, UserLoginExistsException {
+    public UserLogin addLogin(final User user, final UserLogin login) throws UserLoginExistsException, EntityNotFoundException {
         try {
-            uniqueLogin.reserve(login.getUserId(), login.getProvider());
-        } catch (UniqueConstraintException e) {
-            throw new UserLoginExistsException(e.getMessage());
-        }
-
-        Transaction tx = Datastore.beginTransaction();
-        User db = Datastore.getOrNull(tx, userMeta, user.getId());
-        if (db == null) {
-            tx.rollback();
-            uniqueLogin.release(login.getUserId(), login.getProvider());
-            throw new EntityNotFoundException();
+            return TransactionOperator.execute(new ModelOperation<UserLogin>() {
+                @Override
+                public UserLogin execute(com.google.appengine.api.datastore.Transaction tx) throws UserLoginExistsException, EntityNotFoundException {
+                    try {
+                        uniqueLogin.reserveInCurrentTransaction(login.getUserId(), login.getProvider());
+                    } catch (UniqueConstraintException e) {
+                        throw new UserLoginExistsException(e.getMessage());
+                    }
+
+                    User db = Datastore.getOrNull(tx, userMeta, user.getId());
+                    if (db == null) {
+                        uniqueLogin.releaseInCurrentTransaction(login.getUserId(), login.getProvider());
```

Comment:
This is not really needed, it was here before because the reserve wasn't in the same transaction.